app/controllers/registrations_controller.rb

```diff
@@ -154,27 +154,15 @@ def update
 
   private
 
-  # check if we need password to update user data
-  # ie if password or email was changed
-  # extend this as needed
-  def needs_password?(user)
-    user.email != update_params[:email] || update_params[:password].present?
-  end
```

Comment:
I think we may need a modified version of this for when we want to change user pasword.

Reply:
We might be okay. I think `if params[:skip_personal_details] == 'true'` might always evaluate to true (and thus always execute `do_update_password(current_user, update_params)`) when the user is attempting to update a password. I tried to address this a bit in the following comment: https://github.com/portagenetwork/roadmap/pull/917/files#r1783544984